src/rpc/rawtransaction.cpp

```diff
@@ -578,9 +580,8 @@ static RPCHelpMan decodescript()
 
     const std::string type{find_value(r, "type").get_str()};
 
-    if (type == "scripthash") {
-        // P2SH cannot be wrapped in a P2SH. If this script is already a P2SH,
-        // don't return the address for a P2SH of the P2SH.
+    if (type == "scripthash" || type == "witness_v1_taproot") {
```

Comment:
In fa1c1f417ef97d15331f4a13b40be9e00ecdb464 "rpc: Avoid returning P2SH for witness_v1_taproot in decodescript"

I think it would be more future proof to disallow P2SH wrapping for witness v1+ scripts. If this did `IsWitnessProgram` and checked the version was not 0, then this could be futureproofed. Otherwise this list would need to be updated for every new witness version). Doing it in this way would also allow passing a higher witness version script to `decodescript` and not get a p2sh wrapped result.

Reply:
Good point. It is unlikely that a future witness version will be wrappable, so the least maintenance effort would be to disallow all except v0.

Reply:
I am thinking about also skipping P2SH for OP_RETURN data outputs

Reply:
> I am thinking about also skipping P2SH for OP_RETURN data outputs

I think that is reasonable as OP_RETURN in a P2SH is also unspendable.

Reply:
Thanks, done.